cli/engine/providers/azure/InfrastructureBuilder.py

```diff
@@ -24,6 +24,23 @@ def __init__(self, docs, manifest_docs=[]):
         self.docs = docs
         self.manifest_docs = manifest_docs
 
+        # Check if there is a hostname_domain_extension we want to apply.
+        self.hostname_domain_extension = None
+        if self.cluster_model.specification.cloud.hostname_domain_extension != '':
+            self.hostname_domain_extension = self.cluster_model.specification.cloud.hostname_domain_extension
+
+        # Check if there is a hostname_domain_extension we already applied and we want to retain.
+        # The same as VM images we want to preserve hostname_domain_extension over versions.
+        manifest_cluster_model = select_first(self.manifest_docs, lambda x: x.kind == 'epiphany-cluster')
+        if self.manifest_docs:
+            if 'hostname_domain_extension' in manifest_cluster_model.specification.cloud:
+                if manifest_cluster_model.specification.cloud.hostname_domain_extension != '':
```

Comment:
Connected to previous comments, could be removed

Reply:
Cannot as we need to check for hostname preservation. Same as VM images we don't want people to change this value over different versions and/or applies. manifest_cluster_model is coming from the build folder. And as this can be on old cluster which might not have this setting, we need to check for presence, then value, then with the new input.

Reply:
So we can check for presence, but if old value is empty string, why can't we do following steps?

Reply:
Yeah you are right, fixing it.